packages/artillery/lib/platform/aws-ecs/legacy/run-cluster.js

```diff
@@ -1170,35 +1170,38 @@ async function ensureTaskExists(context) {
         };
       });
 
+    const artilleryContainerDefinition = {
+      name: 'artillery',
+      image: imageUrl,
+      cpu: cpu,
+      command: [],
+      entryPoint: ['/artillery/loadgen-worker'],
+      memory: memory,
+      secrets: secrets,
+      ulimits: ulimits,
+      essential: true,
+      logConfiguration: {
+        logDriver: 'awslogs',
+        options: {
+          'awslogs-group': `${context.logGroupName}/${context.clusterName}`,
+          'awslogs-region': context.region,
+          'awslogs-stream-prefix': `artilleryio/${context.testId}`,
+          'awslogs-create-group': 'true',
+          mode: 'non-blocking'
+        }
+      }
+    };
+
     let taskDefinition = {
       family: context.taskName,
-      containerDefinitions: [
-        {
-          name: 'artillery',
-          image: imageUrl,
-          cpu: cpu,
-          command: [],
-          entryPoint: ['/artillery/loadgen-worker'],
-          memory: memory,
-          secrets: secrets,
-          ulimits: ulimits,
-          essential: true,
-          logConfiguration: {
-            logDriver: 'awslogs',
-            options: {
-              'awslogs-group': `${context.logGroupName}/${context.clusterName}`,
-              'awslogs-region': context.region,
-              'awslogs-stream-prefix': `artilleryio/${context.testId}`,
-              'awslogs-create-group': 'true',
-              mode: 'non-blocking'
-            }
-          }
-        },
-        ...([context.adot?.taskDefinition] || [])
```

Comment:
This would always add `undefined` to the resulting array, which means that the check on [line 1204](https://github.com/artilleryio/artillery/pull/3219/files#diff-6ffb0c78caac19b3553805ce01e10b97aea363c2862007d099d09100d91c0150L1204) would always evaluate to `true`.